src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -218,13 +213,96 @@ public Task<ImmutableArray<CodeFixCollection>> GetFixesAsync(Document document,
             return result.ToImmutableAndFree();
         }
 
-        public async Task<CodeFixCollection> GetFixesAsync(Document document, TextSpan range, string diagnosticId, CancellationToken cancellationToken)
+        public async Task<CodeFixCollection> GetFixesInProcAsync(Document document, TextSpan range, string diagnosticId, CancellationToken cancellationToken)
         {
-            var fixesCollectionArray = await GetFixesAsync(document, range, includeSuppressionFixes: false, diagnosticId, cancellationToken).ConfigureAwait(false);
+            var diagnostics = (await _diagnosticService.GetDiagnosticsForSpanAsync(document, range, includeSuppressedDiagnostics:false,  diagnosticId, cancellationToken: cancellationToken).ConfigureAwait(false)).ToList();
+            if (diagnostics.Count == 0)
+            {
+                return null;
+            }
+
+            bool hasAnySharedFixer = _workspaceFixersMap.TryGetValue(document.Project.Language, out var fixerMap);
```

Comment:
isn't code below here all duplication of other method? can we refactor these so that the code here is not just copy and paste from other code in this file?

Reply:
👍